lib/openstudio-standards/standards/necb/NECB2011/building_envelope.rb

```diff
@@ -726,44 +728,116 @@ def apply_max_fdwr_nrcan(model:, fdwr_lim:)
     return true
   end
 
-  # This method is similar to the 'apply_max_fdwr' method above but applies the maximum skylight to roof area ratio to a
-  # building as per NECB 2011 8.4.4.3 and 3.2.1.4 (or equivalent in other versions of the NECB).  It first checks for all
-  # exterior roofs adjacent to conditioned spaces.  It distinguishes between plenums and other conditioned spaces.  It
-  # uses only the non-plenum roof area to calculate the maximum skylight area to be applied to the building.
-  def apply_max_srr_nrcan(model:, srr_lim:)
-    # First determine which roof surfaces are adjacent to heated spaces (both plenum and non-plenum).
-    exp_surf_info = find_exposed_conditioned_roof_surfaces(model)
-    # If the non-plenum roof area is very small raise a warning.  It may be perfectly fine but it is probably a good
-    # idea to warn the user.
-    if exp_surf_info['exp_nonplenum_roof_area_m2'] < 0.1
-      OpenStudio.logFree(OpenStudio::Info, 'openstudio.model.Model', 'This building has no exposed ceilings adjacent to spaces that are not attics or plenums.  No skylights will be added.')
-      return false
-    end
+  # This method is similar to the 'apply_max_fdwr' method above, but applies a maximum skylight-to-roof-area-ratio (SRR)
+  # to a building model, as per NECB 2011 8.4.4.3 and 3.2.1.4 (or equivalent in other NECB vintages). There are 2 options:
+  #
+  # OPTION A: Default, initial BTAP solution. It first checks for all exterior roofs adjacent to conditioned spaces.
+  #           It distinguishes between plenums and other conditioned spaces. It uses only the non-plenum roof area to
+  #           calculate the maximum skylight area to be applied to the building.
+  #
+  # OPTION B: Selected if srr_opt == 'osut'. OSut's 'addSkylights' attempts to meet the requested SRR% target, even if
```

Comment:
Method _apply_max_srr_nrcan_ now offers two options. By default, _OSut_'s _addSkylights_ is not called (`srr_opt:` is defaulted to an empty string). If instead set to `'osut'`, BTAP would rely on _addSkylights_ to match the requested `:srr_lim`, either to match NECB fixed SRR% requirements, or some user-set SRR%.